nuxeo-features/nuxeo-search/nuxeo-search-core/src/main/java/org/nuxeo/ecm/platform/search/core/SavedSearchServiceImpl.java

```diff
@@ -0,0 +1,163 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Gabriel Barata <gbarata@nuxeo.com>
+ */
+package org.nuxeo.ecm.platform.search.core;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Map;
+
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.ACL;
+import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.ecm.platform.query.api.PageProviderService;
+import org.nuxeo.ecm.platform.userworkspace.api.UserWorkspaceService;
+import org.nuxeo.runtime.api.Framework;
+
+import com.google.common.base.Strings;
+
+/**
+ * @since 8.3
+ */
+public class SavedSearchServiceImpl implements SavedSearchService {
+
+    @Override
+    public SavedSearch createSavedSearch(CoreSession session, String title, String queryParams,
+            Map<String, String> namedParams, String query, String queryLanguage, String pageProviderName,
+            Long pageSize, Long currentPageIndex, Long maxResults, String sortBy, String sortOrder,
+            String contentViewData) throws InvalidSearchParameterException, IOException {
+        if (Strings.isNullOrEmpty(title)) {
+            throw new InvalidSearchParameterException("title cannot be empty");
+        }
+
+        if ((!Strings.isNullOrEmpty(query) || !Strings.isNullOrEmpty(queryLanguage))
+                && !Strings.isNullOrEmpty(pageProviderName)) {
+            throw new InvalidSearchParameterException("query and page provider parameters are mutually exclusive"
+                    + " (query, queryLanguage, pageProviderName)");
+        }
+
+        if (Strings.isNullOrEmpty(query) && Strings.isNullOrEmpty(queryLanguage)
+                && Strings.isNullOrEmpty(pageProviderName)) {
+            throw new InvalidSearchParameterException("query or page provider parameters are missing"
+                    + " (query, queryLanguage, pageProviderName)");
+        }
+
+        if (!Strings.isNullOrEmpty(query) && Strings.isNullOrEmpty(queryLanguage)) {
+            throw new InvalidSearchParameterException("queryLanguage parameter is missing");
+        }
+
+        if (Strings.isNullOrEmpty(query) && !Strings.isNullOrEmpty(queryLanguage)) {
+            throw new InvalidSearchParameterException("query parameter is missing");
+        }
+
+        UserWorkspaceService userWorkspaceService = Framework.getLocalService(UserWorkspaceService.class);
```

Comment:
`getLocalService` is deprecated (although not formally), use the identical `getService` instead.